presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduMetadata.java

```diff
@@ -375,7 +418,48 @@ public boolean usesLegacyTableLayouts()
     public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
     {
         KuduTableHandle handle = (KuduTableHandle) table;
-        return new ConnectorTableProperties(handle.getConstraint(), Optional.empty(), Optional.empty(), Optional.empty(), ImmutableList.of());
+        KuduTable kuduTable = handle.getTable(clientSession);
+
+        Optional<ConnectorTablePartitioning> tablePartitioning = Optional.empty();
+        Optional<Set<ColumnHandle>> partitioningColumns = Optional.empty();
+        List<LocalProperty<ColumnHandle>> localProperties = ImmutableList.of();
+
+        if (isKuduGroupedExecutionEnabled(session)
```

Comment:
Do we need this check on groupedExecution here ?

Reply:
If we remove the check it will use grouped_execution strategy always once one/two table satisfied the condition of grouped_execution. So I suggest we keep this parameter as a switch.   :)